curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-pyramid-generator/660f0da9bf1035c9097af20a.md

```diff
@@ -0,0 +1,55 @@
+---
+id: 660f0da9bf1035c9097af20a
+title: Step 23
+challengeType: 0
+dashedName: step-23
+---
+
+# --description--
+
+You should have seen `"freeCodeCamp"` printed to the console. This is because `.pop()` returns the value that was removed from the array - and you pushed `"freeCodeCamp"` to the end of the array earlier.
+
+But what does `.push()` return? Assign your existing `rows.push()` to a new `pushed` variable, and log it.
+
+# --hints--
+
+You should declare a `pushed` variable.
+
+```js
+assert.match(code, /pushed/);
+```
+
+You should use `let` to declare your `pushed` variable.
+
+```js
+assert.match(code, /let\s+pushed/);
+```
+
+You should assign `rows.push("freeCodeCamp")` to your `pushed` variable.
+
+```js
+assert.equal(pushed, 4);
+```
+
+You should log your `pushed` variable. Don't forget your semi-colon.
+
+```js
+assert.match(code, /console\.log\(\s*pushed\s*\)/);
+```
```

Comment:
this also doesn't check for semi colons but they should be optional anyway

````suggestion
You should log your `pushed` variable.

```js
assert.match(code, /console\.log\(\s*pushed\s*\)/);
```
````